documentation/ci-fix.md

```diff
@@ -126,20 +126,11 @@ Refer to [Semantic and Model Violations Reference](https://github.com/Azure/azur
 
 ## `Swagger BreakingChange` and `BreakingChange(Cross-Version)`
 
-- An API contract is identified by its api-version value. Once published, no changes to this API contract are allowed. This applies regardless of whether the API contract is for private preview, public preview, or GA (stable).
-    - The same-version breaking change linter rules check for changes to an existing api-version OpenAPI spec.
-        - When introducing a new API contract (preview or not), the new API contract must be backwards compatible with the previous GA’s API contract.
-            - However, during a (private or public) preview cycle, a new preview API contract does not have to be backwards compatible with the previous preview API contract although it must still be backwards compatible with the latest GA API contract.
-            - The cross version breaking change linter rules checks for this by comparing the new OpenAPI spec with the latest GA OpenAPI spec. If there is no latest GA OpenAPI spec, then the latest preview if it > 1 year old. If nether a GA or preview > 1 year old exists, then the OpenAPI spec is considered good.
-
-### Adding label on PR automatically
-
-The breaking change check has two types of violations: one is breaking change in the same version but not breaking change in a new version, the other is breaking change even in a new version.
-For the former, a label 'NewApiVersionRequired' will be added automatically; For the latter, a label 'BreakingChangeReviewRequired' will be added automatically. Adding each label will trigger a github comment with guldance on how to fix.
+See [aka.ms/azsdk/pr-brch-deep](https://aka.ms/azsdk/pr-brch-deep). If you want a quick read, see only [the `summary` section](https://aka.ms/azsdk/pr-brch-deep#summary).
```

Comment:
@konrad-jamrozik @kurtzeborn is there an updated Public source for this guidance? We're seeing issues where Service Teams are regularly breaking this guidance already (e.g. https://github.com/Azure/azure-rest-api-specs/issues/28380, from less than a week ago) so I'm concerned that removing the Public source for this is only going to exacerbate this issue unfortunately

Reply:
@tombuildsstuff the linked doc should be available to anyone withing Microsoft. Is this not the case?

Reply:
@konrad-jamrozik unfortunately there's a considerable number of folks interacting with this repository who don't work at Microsoft and yet need to reference that document to raise issues with Service Teams.

Whilst I appreciate that your team may not rely on this information, I've found that the breaking change detector is pretty broken (see links below) - so given that external users contribute to this repository, there needs to be a public reference for this; in addition to whatever automated tooling is used.

* Bug in the Breaking Change detector: https://github.com/Azure/azure-rest-api-specs/issues/26304
* Adding a new optional QueryString parameter isn't a breaking change: https://github.com/Azure/azure-rest-api-specs/pull/25080 (it _could_ be to the SDK, but this is the API definition rather than the SDK being changed)
* Documenting an optional field that exists in the API that isn't described in the Swagger https://github.com/Azure/azure-rest-api-specs/pull/22407
* Making the URI segment parameters consistent: https://github.com/Azure/azure-rest-api-specs/pull/21667 and https://github.com/Azure/azure-rest-api-specs/pull/20588 - I'm surprised there isn't a linter for this
* Making the URI segment consistent (some are a String some are an Enum, but support both values): https://github.com/Azure/azure-rest-api-specs/pull/23562
* Updating the maximum value for an existing parameter: https://github.com/Azure/azure-rest-api-specs/pull/15619
* This one is _technically_ a breaking change, but it's not in a practical sense: https://github.com/Azure/azure-rest-api-specs/pull/26838 (anyone using the API will be setting one of these values, therefore changing this from a String -> Enum is _technically_ a breaking change since code would require changes - but isn't in real terms/the API behaviour)

Reply:
@tombuildsstuff

Re:

> Adding a new optional QueryString parameter isn't a breaking change: https://github.com/Azure/azure-rest-api-specs/pull/25080 (it could be to the SDK, but this is the API definition rather than the SDK being changed)

I see:

https://github.com/Azure/azure-rest-api-specs/pull/25080/checks

![image](https://github.com/Azure/azure-rest-api-specs/assets/4429827/54539abd-a022-4165-a6a7-9a9d235d9199)

And the PR got `VersioningReviewRequired` which means this PR possibly violates Azure Versioning Policy due to compatible (non-breaking) change.

In general, you made some statements about what is or what is not a breaking change, but I suspect you are unaware of the Azure Versioning Policy that dictates what is flagged by the tooling or not. This policy may not be in agreement with your assessments.

Regarding the need for docs becoming public: do you perhaps have an issue you could point me to where you elaborate on your use case? This will help my team to evaluate which docs to make public.

Reply:
@JeffreyRichter in case you are interested, this thread has more feedback from @tombuildsstuff .

Reply:
FYI for @tombuildsstuff (and whoever else might read this)...
Azure has both versioning and breaking change policies.

ANY change to an API contract REQUIRES a new api-version value. So, adding a new optional parameter is not breaking but a new api-version is still required. Therea re multiple reasons for this policy but one reason is because a might need to know what api-version supports the new optional parameter and if the api-version doesn't change, then there is no way to know if it is supported on their specific cloud.

Azure's breaking change policy goes like this: Customers should be able to adopt a new api-version of a service without requiring ANY other code changes to get the same exact behavior they got before. If a code change is REQUIRED, then the customer is broken. Some changes will not break the HTTP API but may break SDKs - in this case, we still consider them breaking as a large set of Azure customers use our SDKs.